api/metrics/renderers.py

```diff
@@ -64,10 +76,53 @@ def render(self, json_response, accepted_media_type=None, renderer_context=None)
             csv_writer.writerow(
                 get_csv_row(csv_fieldnames, serialized_report),
             )
+
+        response = renderer_context['response']
+        filename = self.get_filename(renderer_context, self.extension)
+        response['Content-Disposition'] = f'attachment; filename="{filename}"'
+
         return csv_filecontent.getvalue()
 
 
-class MetricsReportsTsvRenderer(MetricsReportsCsvRenderer):
+class MetricsReportsTsvRenderer(MetricsReportsRenderer):
     format = 'tsv'
+    extension = 'tsv'
     media_type = 'text/tab-separated-values'
     CSV_DIALECT = csv.excel_tab
+
+
+class MetricsReportsCsvRenderer(MetricsReportsRenderer):
+    format = 'csv'
+    extension = 'csv'
+    media_type = 'text/csv'
+    CSV_DIALECT = csv.excel
+
+
+class MetricsReportsJsonRenderer(MetricsReportsRenderer):
+    """
+    Just the the basic report without nested user details.
+    """
+    format = 'json_report'
+    extension = 'json'
+    media_type = 'application/json'
+
+    def render(self, json_response, accepted_media_type=None, renderer_context=None):
+        response = renderer_context['response']
+        filename = self.get_filename(renderer_context, self.extension)
+        response['Content-Disposition'] = f'attachment; filename="{filename}"'
+        return json.dumps([item['attributes'] for item in json_response['data']])
+
+
+class MetricsReportsJsonDirectDownloadRenderer(MetricsReportsRenderer):
+    """
+    The whole raw report with pagination and filtering/sorts
+    """
+    format = 'direct_download'
+    extension = 'json'
+    media_type = 'application/json'
+
+    def render(self, data, accepted_media_type=None, renderer_context=None):
+        response = renderer_context['response']
+        filename = self.get_filename(renderer_context, self.extension)
+        response['Content-Disposition'] = f'attachment; filename="{filename}"'
+        return json.dumps(data)
```

Comment:
this renderer convinces me that setting `Content-Disposition` is a view concern, not a renderer concern -- i gather this is meant to be "same jsonapi rendering as the rest of the api, but downloaded", but it's missing logic from the common `JSONAPIRenderer` (and inheritance is not the answer)

still thinking an additional query param (handled by a base view) to make any response a download would be a less complicated way to go